core/src/Kokkos_Macros.hpp

```diff
@@ -652,4 +652,13 @@
 #define KOKKOS_THREAD_LOCAL __thread
 #endif
 
+#define KOKKOS_PROFILING_STRINGIFY(x) #x
```

Comment:
@AndrewGaspar, are you comfortable giving this a once over? I don't know whether this macro implementation is how stuff should be done. I know that it works.

Essentially, if I have defined KOKKOS_PROFILING_INCLUDE_SOURCE_INFO, I want 

`KOKKOS_PROFILING_NAME(dogs)` to yield `file.cpp:22:dogs`, but if that macro isn't defined, just yield `dogs`. I know it's functional, I just want to make sure it isn't against proper practice for defining macros

Reply:
It takes days of immersion for my brain to speak C macros, so this might be a naive question, but why do you need both `KOKKOS_PROFILING_STRINGIFY` and `KOKKOS_PROFILING_TOSTRING`?

Reply:
Anyway, this seems fine to me from a design perspective.

Reply:
Yes you need the two levels if you want to expand `__LINE__`